scripts/externalTests/update_external_repos.sh

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+
+#------------------------------------------------------------------------------
+# Clones external test repositories from solidity-external-tests organization
+# and for each of them pulls latest upstream changes from the main branch and
+# pushes them to our fork.
+#
+# The script assumes that the current user has write access to
+# solidity-external-tests and that git is configured to be able to push there
+# without specifying the password (e.g. with the key already unlocked and loaded
+# into ssh-agent). Otherwise git will keep asking for password for each repository.
+#
+# Usage:
+#
+#    ./update_external_repos.sh [<target_dir>]
+#
+# <target_dir>: directory where the clones of the repositories are stored.
+#               If omitted, a temporary directory will be created.
+#
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2022 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -euo pipefail
+
+target_dir="${1:-$(mktemp -d -t update_external_repos_XXXXXX)}"
```

Comment:
Don't we want to remove a temporary directory at the end of the script?

Reply:
Not necessarily. You might want to inspect contents or rerun the script, now giving it that tmp path. It's a relatively heavy operation so I prefer to err on the side of not removing it automatically. My assumption is that the temp dir does not really hurt being there for a short time and it will be cleared anyway on next reboot (or whenever the user cleans his tmp if it's not on tmpfs).